src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -197,25 +207,29 @@ private static boolean editDistanceIsOne(final String incomingEdgeName,
     private static boolean identifierSubstringsAreEqual(final String incomingEdgeName,
             final String startingEdgeName)
     {
+
         // Gather identifier substrings from both road names
-        final List<String> incomingEdgeNameAlphanumericIdentifierStrings = Arrays
+        final List<String> incomingEdgeNameAlphanumericIdentifierStrings;
+        incomingEdgeNameAlphanumericIdentifierStrings = Arrays
                 .stream(incomingEdgeName.split(WHITESPACE_REGEX))
-                .filter(substring -> substring.matches(ALPHANUMERIC_IDENTIFIER_STRING_REGEX)
-                        || substring.matches(CHARACTER_IDENTIFIER_STRING_REGEX)
-                        || Stream.of(CJKNumbers.values())
-                                .anyMatch(cjkNumber -> substring.contains(
-                                        new String(Character.toChars(cjkNumber.getValue())))))
+                .filter(substring -> Stream.of(CJKNumbers.values())
+                        .anyMatch(cjkNumber -> substring
+                                .contains(new String(Character.toChars(cjkNumber.getValue()))))
+                        || ALPHANUMERIC_IDENTIFIER_STRING_PATTERN.matcher(substring).matches()
+                        || CHARACTER_IDENTIFIER_STRING_PATTERN.matcher(substring).matches())
                 .collect(Collectors.toList());
+
         final List<String> startingEdgeNameAlphanumericIdentifierStrings = Arrays
                 .stream(startingEdgeName.split(WHITESPACE_REGEX))
-                .filter(substring -> substring.matches(ALPHANUMERIC_IDENTIFIER_STRING_REGEX)
-                        || substring.matches(CHARACTER_IDENTIFIER_STRING_REGEX)
-                        || Stream.of(CJKNumbers.values())
-                                .anyMatch(cjkNumber -> substring.contains(
-                                        new String(Character.toChars(cjkNumber.getValue())))))
+                .filter(substring -> Stream.of(CJKNumbers.values())
```

Comment:
Pretty nit, but the ```Stream.of(CJKNumbers.values()).anyMatch(cjkNumber -> substring.contains(new String(Character.toChars(cjkNumber.getValue()))))``` might be better off after the other two conditional checks (alphanumeric identifier, character identifier) since most data is probably not in ChineseJapaneseKorean languages

Reply:
Okay. Initially we thought it was regex thats taking time. So, I put regex conditions after the other checks so that if other checks fail we don't even have to go to regex.

Reply:
Do you want it to be coded like below?

final List<String> startingEdgeNameAlphanumericIdentifierStrings = Arrays
                .stream(startingEdgeName.split(WHITESPACE_REGEX))
                .filter(substring -> substring.matches(ALPHANUMERIC_IDENTIFIER_STRING_REGEX)
                        || substring.matches(CHARACTER_IDENTIFIER_STRING_REGEX)
                        || Stream.of(CJKNumbers.values())
                                .anyMatch(cjkNumber -> substring.contains(
                                        new String(Character.toChars(cjkNumber.getValue())))))
                .collect(Collectors.toList());

Reply:
Yup, that would be the better way since we don't need to check for CJK necessarily every time.

Reply:
Sorry I was unclear -- same thing for lines 212-219!

Reply:
Done